### PR TITLE
Draft for multi PVN support + protocol extension

### PR DESCRIPTION
**Pull request by Moresteck**

### CODE REVIEW

**Review by notdevcody — CHANGES_REQUESTED** [review]

File: src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java [hunk 1]
```diff
@@ -99,7 +99,8 @@ public void setHandle(final EntityPlayer entity) {
 
     public void sendRawMessage(String message) {
     	// uberbukkit - fix worldedit cui
-    	if (Uberbukkit.getPVN() < 11 && message.equals("\u00A75\u00A76\u00A74\u00A75")) return;
+    	int pvn = getHandle().netServerHandler.networkManager.pvn;
+    	if (pvn < 11 && message.equals("\u00A75\u00A76\u00A74\u00A75")) return;
```
[the remark this answers is not in the document]
Comment (Moresteck) [reply src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java:103]: It's not necessary to know what these are. It's the character sequence WorldEdit plugin greets the client with, for it to later communicate back with the plugin. The sequence crashes versions before b1.5, thus why it's being filtered out.

File: src/main/java/net/minecraft/server/Packet62Sound.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import java.io.DataOutputStream;
+import java.io.IOException;
+
+public class Packet62Sound extends Packet {
+	public String sound;
+	public double locX;
+	public double locY;
+	public double locZ;
+	public float f;
+	public float f1;
```
[the remark this answers is not in the document]
Comment (Moresteck) [reply src/main/java/net/minecraft/server/Packet62Sound.java:13]: `f` should be `volume`, and `f1` should become `pitch`, as per MCP mappings

File: src/main/java/net/minecraft/server/Packet2Handshake.java [hunk 3]
```diff
@@ -4,30 +4,38 @@
 import java.io.DataOutputStream;
 import java.io.IOException;
 
-import pl.moresteck.uberbukkit.Uberbukkit;
-
 public class Packet2Handshake extends Packet {
 
     public String a;
+    public boolean pvn11;
```
[the remark this answers is not in the document]
Comment (Moresteck) [reply src/main/java/net/minecraft/server/Packet2Handshake.java:10]: It's impossible to deduct the exact PVN by just reading this packet. We can only know the PVN range the client is in. What better name should the field have? Maybe `isPVN11OrHigher`? Doesn't look too good tbh.

**Review by notdevcody — APPROVED** [review]
LGTM

**Review by notdevcody — CHANGES_REQUESTED** [review]

**Review by notdevcody — CHANGES_REQUESTED** [review]

**Review by notdevcody — APPROVED** [review]
Looks like ready to merge.

**Pull request merged**